core/src/avm1/property.rs

```diff
@@ -52,23 +52,16 @@ impl<'gc> Property<'gc> {
 
     /// Set a property slot.
     ///
-    /// This function returns the `ReturnValue` of the property's virtual
+    /// This function may return an `Executable` of the property's virtual
     /// function, if any happen to exist. It should be resolved, and it's value
     /// discarded.
-    pub fn set(
-        &mut self,
-        activation: &mut StackFrame<'_, 'gc>,
-        context: &mut UpdateContext<'_, 'gc, '_>,
-        this: Object<'gc>,
-        base_proto: Option<Object<'gc>>,
-        new_value: impl Into<Value<'gc>>,
-    ) -> Result<ReturnValue<'gc>, Error<'gc>> {
+    pub fn set(&mut self, new_value: impl Into<Value<'gc>>) -> Option<Executable<'gc>> {
         match self {
             Property::Virtual { set, .. } => {
                 if let Some(function) = set {
-                    function.exec(activation, context, this, base_proto, &[new_value.into()])
+                    Some(function.to_owned())
```

Comment:
TODO for a later PR: This is a fairly chunky clone (`Executable` is 128+ bytes). Can we get rid of or minimize this clone while still fixing the double borrow? (One easy thing to do is to replace the block of bools in Avm1Function with an `EnumSet` save some data).

Reply:
Maybe we just cram `Avm1Function` in a `Gc`.